nav2_costmap_2d/plugins/inflation_layer.cpp

```diff
@@ -296,11 +298,9 @@ InflationLayer::updateCosts(
         enqueue(index + size_x, mx, my + 1, sx, sy);
       }
     }
-  }
-
-  for (auto & dist : inflation_cells_) {
-    dist.clear();
-    dist.reserve(200);
+    // This level of inflation_cells_ is not needed anymore. We can free the memory
+    // Note that dist_bin.clear() is not enough, because it won't free the memory
+    dist_bin = std::vector<CellData>();
```

Comment:
Out of curiosity, this is the real fix, right?

Reply:
I was thinking the same

Reply:
note has the cleaning is done inside the for loop iterating through `inflation_cells_` and not later

Reply:
also the time at which we `reserve` the memory is different